### PR TITLE
Christy/react editing

### PR DESCRIPTION
**Pull request by ChristyReynolds**
Issue 148: setting persistent scaling for Stokes channels

### CODE REVIEW

File: src/components/RenderConfig/RenderConfigComponent.tsx [hunk 1]
```diff
@@ -36,8 +36,8 @@ export class RenderConfigComponent extends React.Component<WidgetProps> {
     private cachedFrame: FrameStore;
 
     @observable width: number;
-    @observable height: number;
-
+    @observable height: number;
```
Comment (veggiesaurus) [inline src/components/RenderConfig/RenderConfigComponent.tsx]: whitespace typo

File: src/components/RenderConfig/RenderConfigComponent.tsx [hunk 2]
```diff
@@ -123,13 +123,15 @@ export class RenderConfigComponent extends React.Component<WidgetProps> {
 
     handleScaleMinChange = (val: number) => {
         if (isFinite(val)) {
-            this.props.appStore.activeFrame.renderConfig.setCustomScale(val, this.props.appStore.activeFrame.renderConfig.scaleMax);
+            const stokes = this.props.appStore.activeFrame.renderConfig.stokes;
```
Comment (veggiesaurus) [inline src/components/RenderConfig/RenderConfigComponent.tsx]: We don't need this variable anymore, do we? Can't see where it's being used

File: src/components/RenderConfig/RenderConfigComponent.tsx [hunk 3]
```diff
@@ -123,13 +123,15 @@ export class RenderConfigComponent extends React.Component<WidgetProps> {
...
+            this.props.appStore.activeFrame.renderConfig.setCustomScale(val, this.props.appStore.activeFrame.renderConfig.scaleMaxVal);
         }
     };
 
     handleScaleMaxChange = (val: number) => {
         if (isFinite(val)) {
-            this.props.appStore.activeFrame.renderConfig.setCustomScale(this.props.appStore.activeFrame.renderConfig.scaleMin, val);
+            const stokes = this.props.appStore.activeFrame.renderConfig.stokes;
```
Comment (veggiesaurus) [inline src/components/RenderConfig/RenderConfigComponent.tsx]: Same as above

File: src/components/RenderConfig/RenderConfigComponent.tsx [hunk 4]
```diff
@@ -306,7 +307,7 @@ export class RenderConfigComponent extends React.Component<WidgetProps> {
             percentileSelectDiv = (
                 <div className="percentile-select">
                     <FormGroup label="Clip Percentile" inline={true}>
-                        <HTMLSelect options={percentileRankOptions} value={frame.renderConfig.selectedPercentile} onChange={this.handlePercentileRankSelectChanged}/>
+                        <HTMLSelect options={percentileRankOptions} value={frame.renderConfig.selectedPercentile[stokes]} onChange={this.handlePercentileRankSelectChanged}/>
```
Comment (veggiesaurus) [inline src/components/RenderConfig/RenderConfigComponent.tsx]: suggestion: we should make a @computed value `selectedPercentileVal` similar to how you've handled the `scaleMinVal` computed values

**Comment by veggiesaurus** [comment]
@kswang1029 can you test this using mix 'n match? To verify it solves #148

**Comment by kswang1029** [comment]
@ChristyReynolds Nice work! 
One issue:
with the default 99.9% clip, the x-range of the plot changes with different Stokes (which is good). However, if the x-range of a given Stokes is changed by users via mouse interactions, then the x-range is fixed to all Stokes. I encountered a case that I zoom in the Stokes=0, then switch to Stokes=1, and the plot is empty. Since it is empty there is no way to use mouse db-click to reset the range. I will need to use the reset range button in the settings menu to cure this. I suggest each Stokes should have its x-range definition. Sounds reasonable? @ChristyReynolds @veggiesaurus

**Comment by veggiesaurus** [comment]
@kswang1029 the same issue occurs if you're zoomed in and change to another channel with a different histogram range. Perhaps we should just reset the histogram range whenever the histogram itself changes? (so, switching channel, switching channel <-> cube or switching Stokes)

**Comment by kswang1029** [comment]
Hmm, right. Let's reset the range when switching channel/cube or Stokes.

**Comment by veggiesaurus** [comment]
> Hmm, right. Let's reset the range when switching channel/cube or Stokes.

@kswang1029 Can you add a new issue for this? It's not directly dependent on this PR, so I'll merge this in and we can look into the issue separately.

**Comment by kswang1029** [comment]
sure. Other than that I think this PR works well.

**Pull request merged**